devolutions-gateway/src/lib.rs

```diff
@@ -77,6 +81,13 @@ impl DgwState {
         let (shutdown_handle, shutdown_signal) = devolutions_gateway_task::ShutdownHandle::new();
         let (job_queue_handle, job_queue_rx) = job_queue::JobQueueHandle::new();
         let credential_store = credential::CredentialStoreHandle::new();
+        
+        let temp_dir = TempDir::new("dgw-network-monitor-test").expect("Could not create temp dir");
+        let temp_path: Utf8PathBuf = Utf8Path::from_path(temp_dir.path())
+            .expect("TempDir gave us a garbage path")
+            .to_path_buf();
+        
+        let monitoring_state = Arc::new(network_monitor::State::mock(temp_path));
```

Comment:
thought: I would prefer we do not include an extra dependency (tempdir) just for mocking (testing). I’ll come back on this later.